server/src/main/java/org/opensearch/index/get/ShardGetService.java

```diff
@@ -216,25 +216,18 @@ private GetResult innerGet(
         fetchSourceContext = normalizeFetchSourceContent(fetchSourceContext, gFields);
 
         Term uidTerm = new Term(IdFieldMapper.NAME, Uid.encodeId(id));
-        Engine.GetResult get = indexShard.get(
+
+        try (Engine.GetResult get = indexShard.get(
             new Engine.Get(realtime, true, id, uidTerm).version(version)
                 .versionType(versionType)
                 .setIfSeqNo(ifSeqNo)
                 .setIfPrimaryTerm(ifPrimaryTerm)
-        );
-        if (get.exists() == false) {
-            get.close();
-        }
-
-        if (get == null || get.exists() == false) {
-            return new GetResult(shardId.getIndexName(), id, UNASSIGNED_SEQ_NO, UNASSIGNED_PRIMARY_TERM, -1, false, null, null, null);
-        }
-
-        try {
+        )) {
+            if (get.exists() == false) {
```

Comment:
A small question, does it cover the condition when `get == null`?

Reply:
@tlfeng Thanks for your review, I checked, the `GetResult` will never be null here, but this is a good suggestion, add a check is better.
And I have format the code, please review thanks.

Reply:
Thanks for your time checking the situation! I'm just curious, because I'm not familiar with this part of code, not exactly a suggestion 😄.

Reply:
Oh, added the signed of :).